sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -95,18 +59,12 @@ export interface OperationTracingOptions {
     tracingContext?: Context;
 }
 
-// @public
-export function setSpan(context: Context, span: Span): Context;
-
-// @public
-export function setSpanContext(context: Context, spanContext: SpanContext): Context;
-
 // @public
 export interface Span {
```

Comment:
I love to see the simplification!

Could you lay out what the next sequence of steps you think are? e.g. will Span eventually be a re-export of OT? Do we need to have some minimal interface for it ourselves?

Same with `context` - can we avoid having SDKs interact with context directly or even have to know about it?

Reply:
Sure! 

Assumption: EventHubs and ServiceBus are ok taking a direct dependency on @opentelemetry/api

## Must stay

- SpanStatus, SpanStatusCode
- isSpanContextValid
- Span
- SpanContext, TraceState
- SpanOptions, Link, SpanKind, SpanAttributes, SpanAttribute
- OperationTracingOptions + Context (to support manual span propagation)

I'd love to remove all of these, but these are the ones that will be really hard to tease out at this point

## Can go to azure/test-utils (in this PR) and just re-exported from OTel

- Tracer interface
- ContextAPI
- getTracer
- context entrypoint 
- TraceFlags

## Longer term

- I'd like to avoid having packages manage spans entirely using something like [withTrace](https://github.com/maorleger/azure-sdk-for-js/blob/withtrace-core-tracing/sdk/core/core-tracing/src/createTrace.ts) which is in early draft and likely will take time to migrate packages over.
- I'd also like to re-export OTel interfaces directly but we will likely not get there by GA (there are a few things that have to happen first)

What do you think?

Reply:
> I'd also like to re-export OTel interfaces directly but we will likely not get there by GA (there are a few things that have to happen first)

Figured I'll expand what those are: 
1. Figure out how to downlevel OTel's interfaces: can I even downlevel-dts interfaces I am re-exporting? Alternatively, can we set TS 3.8 as the min-bar for support? 😈 
2. Upgrade API-extractor across the board to something that supports import/export type syntax (otherwise there will be a _ton_ of build noise in the form of API Extractor warnings)

Reply:
This is an interesting point. Since tracing is **heavily** dependent on OpenTelemetry, I think it's reasonable to match their support level for TS. So if they only work on TS 3.8+, I think it's fair that we are the same (for tracing interfaces, that is.) I don't want to be in the business of trying to monkeypatch their compat story.

Reply:
Also I'm curious what the blocker is from re-exporting the OT shapes? Why does our definition of Span vs theirs make a meaningful difference to TS?

Reply:
> Also I'm curious what the blocker is from re-exporting the OT shapes? Why does our definition of Span vs theirs make a meaningful difference to TS?

There's two issues that I am aware of and I believe both are related to `import type` and `export type` being used in [OTel's API](https://github.com/open-telemetry/opentelemetry-js-api/blob/a752e7f814e7b335e50705d281cb2c6e1bd38a35/src/index.ts#L55) 

The first is that our version of API Extractor (7.7.x) does not support them so we end up with these warnings on every package when I try to re-export any APIs directly from OTel:

```
Warning: /home/mleger/workspace/worktrees/tracing/common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/src/index.d.ts:27:8 - (TS2304) Cannot find name 'type'.
Warning: /home/mleger/workspace/worktrees/tracing/common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/src/index.d.ts:27:28 - (TS2304) Cannot find name 'from'.
Warning: /home/mleger/workspace/worktrees/tracing/common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/src/index.d.ts:31:8 - (TS2304) Cannot find name 'type'.
Warning: /home/mleger/workspace/worktrees/tracing/common/temp/node_modules/.pnpm/@opentelemetry+api@1.0.3/node_modules/@opentelemetry/api/build/src/index.d.ts:31:26 - (TS2304) Cannot find name 'from'.
```

See #9410 and #13557 as related issues (and of course upgrading API Extractor is not free because of how it handles name collision in later versions, see #17702 for an example)

The second is less clear to me - I'm hopeful @richardpark-msft or @chradek remember why https://github.com/Azure/azure-sdk-for-js/pull/14618 was needed? That looks like where we removed the re-exports and redeclared the types